maps/reddit.py

```diff
@@ -0,0 +1,172 @@
+    try:
+        dataset = atlas.map_data(data=my_data,
+                                 indexed_field='text',
+                                 description='Reddit comments mapped via automation.',
+                                 topic_model=True
+                                )
+        if dataset and 'id' in dataset:
+            print("Map created on Atlas with ID:", dataset['id'])
+            print("All done! Visit the map link to see the status of your map build.")
+        else:
+            print("Map creation failed. Dataset ID not found in response.")
+    except Exception as e:
+        print(f"An error occurred while building map on Atlas: {e}")
+
+# Function to scrape Reddit comments with retry on rate limit
+def scrape_reddit_comments(reddit, submission):
+    max_attempts = 5  # Maximum retry attempts
+    current_attempt = 0
+    while current_attempt < max_attempts:
+        try:
+            submission.comments.replace_more(limit=None)
+            comments = fetch_all_comments(submission)
+            print(f"Number of comments fetched: {len(comments)}")
+            return comments
+        except PrawcoreException as e:
+            if e.response and e.response.status_code == 429:
+                delay = 2 ** current_attempt  # Exponential backoff
+                print(f"Rate limit exceeded. Retrying in {delay} seconds...")
+                time.sleep(delay)
+                current_attempt += 1
+            else:
+                print(f"Error scraping comments: {e}")
+                return []
+    
+    print("Max retry attempts reached. Could not fetch comments.")
+    return []
+
+# Function to fetch all comments (modified to fit)
+def fetch_all_comments(submission):
+    comments = []
+    submission.comments.replace_more(limit=None)
+    for comment in submission.comments.list():
+        comment_data = {
+            'text': comment.body,
+            'author': comment.author.name if comment.author else '[deleted]',
+            'score': comment.score
+        }
+        comments.append(comment_data)
+        comments.extend(fetch_replies(comment))
+    return comments
+# Function to fetch replies recursively (modified to fit)
+def fetch_replies(comment):
+    replies = []
+    if isinstance(comment, praw.models.reddit.comment.Comment):
+        if hasattr(comment, 'replies') and isinstance(comment.replies, praw.models.comment_forest.CommentForest):
+            comment.replies.replace_more(limit=None)
+            for reply in comment.replies.list():
+                reply_data = {
+                    'text': reply.body,
+                    'author': reply.author.name if reply.author else '[deleted]',
+                    'score': reply.score
+                }
+                replies.append(reply_data)
+                replies.extend(fetch_replies(reply))
+    return replies
+
+# Function to save comments to CSV file (modified to fit)
+def save_to_csv(comments, filename):
```

Comment:
instead of using CSVs as an interchange format, you should use Apache Arrow (which has typing)

Reply:
map_data will directly take an Apache Arrow table as input